src/NUnitFramework/framework/Constraints/DictionaryContainsKeyConstraint.cs

```diff
@@ -65,32 +71,138 @@ public override string Description
         /// </summary>
         protected object Expected { get; }
 
+        /// <summary>
+        /// Flag the constraint to ignore case and return self.
+        /// </summary>
+        [Obsolete("Deprecated, use Does.ContainKey")]
```

Comment:
Given that they have just typed `Does.ContainKey(...).IgnoreCase`, they're going to think, "I already am using `Does.ContainKey`." Could we be more specific about what they need to do if they want to do an case-insensitive key assert, e.g. `Has.Property("Keys").With.One.EqualTo(...).IgnoreCase`?

It would read better to replace the comma with either a semicolon or a period followed by starting a new sentence.